packages/contracts-bedrock/contracts/L1/OptimismPortal.sol

```diff
@@ -371,26 +366,16 @@ contract OptimismPortal is Initializable, ResourceMetering, Semver {
         // Mark the withdrawal as finalized so it can't be replayed.
         finalizedWithdrawals[withdrawalHash] = true;
 
-        // We want to maintain the property that the amount of gas supplied to the call to the
-        // target contract is at least the gas limit specified by the user. We can do this by
-        // enforcing that, at this point in time, we still have gaslimit + buffer gas available.
-        require(
-            gasleft() >= _tx.gasLimit + FINALIZE_GAS_BUFFER,
-            "OptimismPortal: insufficient gas to finalize withdrawal"
-        );
-
         // Set the l2Sender so contracts know who triggered this withdrawal on L2.
         l2Sender = _tx.sender;
 
         // Trigger the call to the target contract. We use SafeCall because we don't
         // care about the returndata and we don't want target contracts to be able to force this
-        // call to run out of gas via a returndata bomb.
-        bool success = SafeCall.call(
-            _tx.target,
-            gasleft() - FINALIZE_GAS_BUFFER,
-            _tx.value,
-            _tx.data
-        );
+        // call to run out of gas via a returndata bomb. We also want to maintain the property
+        // that the amount of gas supplied to the call to the target contract is at least the gas
+        // limit specified by the user. If there is not enough gas in the callframe to accomplish
+        // this, `callWithMinGas` will revert.
+        bool success = SafeCall.callWithMinGas(_tx.target, _tx.gasLimit, _tx.value, _tx.data);
```

Comment:
So now we make the call with at least `_tx.gasLimit`, but do we have enough left to handle the gas clean up, event emission and estimation logic?

Reply:
Hmmm... I guess that doesn't matter because if not then the whole thing OOGs and reverts... 
... right?

Reply:
Exactly- the `callWithMinGas` function doesn't care about this, if there's not enough gas left, the function will revert anyways.